diff --git a/node_modules/react-native-switch/index.d.ts b/node_modules/react-native-switch/index.d.ts
index 72cd110..5d1d381 100644
--- a/node_modules/react-native-switch/index.d.ts
+++ b/node_modules/react-native-switch/index.d.ts
@@ -30,7 +30,8 @@ declare module 'react-native-switch' {
     switchRightPx?: number;
     switchWidthMultiplier?: number;
     switchBorderRadius?: number;
+    useNativeDriver?: boolean
   };
 
-  export class Switch extends Component<SwitchProps> {};
+  export class Switch extends Component<SwitchProps> { };
 }
diff --git a/node_modules/react-native-switch/lib/Switch.js b/node_modules/react-native-switch/lib/Switch.js
index c4ccc20..1856162 100644
--- a/node_modules/react-native-switch/lib/Switch.js
+++ b/node_modules/react-native-switch/lib/Switch.js
@@ -39,7 +39,8 @@ export class Switch extends Component {
     switchLeftPx: PropTypes.number,
     switchRightPx: PropTypes.number,
     switchWidthMultiplier: PropTypes.number,
-    switchBorderRadius: PropTypes.number
+    switchBorderRadius: PropTypes.number,
+    useNativeDriver: PropTypes.bool
   };
 
   static defaultProps = {
@@ -68,7 +69,8 @@ export class Switch extends Component {
     switchRightPx: 2,
     switchWidthMultiplier: 2,
     switchBorderRadius: null,
-    testID: null
+    testID: null,
+    useNativeDriver: false
   };
 
   constructor(props, context) {
@@ -130,19 +132,23 @@ export class Switch extends Component {
       Animated.spring(this.state.transformSwitch, {
         toValue: value
           ? this.props.circleSize / this.props.switchLeftPx
-          : -this.props.circleSize / this.props.switchRightPx
+          : -this.props.circleSize / this.props.switchRightPx,
+        useNativeDriver:this.props.useNativeDriver
       }),
       Animated.timing(this.state.backgroundColor, {
         toValue: value ? 75 : -75,
-        duration: 200
+        duration: 200,
+        useNativeDriver:this.props.useNativeDriver
       }),
       Animated.timing(this.state.circleColor, {
         toValue: value ? 75 : -75,
-        duration: 200
+        duration: 200,
+        useNativeDriver:this.props.useNativeDriver
       }),
       Animated.timing(this.state.circleBorderColor, {
         toValue: value ? 75 : -75,
-        duration: 200
+        duration: 200,
+        useNativeDriver:this.props.useNativeDriver
       })
     ]).start(cb);
   };
